### PR TITLE
introduce sentry to bleep

### PR DESCRIPTION
**Pull request by nerdypepper**
- attempts to load environment variables from `.env`
- attempts to initialize sentry with dsn: `VITE_SENTRY_DSN_BE`
- attempts to initialize segment with key from `SEGMENT_KEY_BE`
- gets rid of cli flag to initialize segment

### CODE REVIEW

**Review by ggordonhall — APPROVED** [review]
Looking good. One thing, how will this interact with the Sentry client initialised at the Tauri level?

File: server/bleep/src/lib.rs [hunk 1]
```diff
@@ -315,6 +331,29 @@ impl Application {
...
+
+        if sentry::Hub::current().client().is_some() {
+            warn!("sentry has already been initialized");
+            return;
+        }
+
+        info!("initializing sentry ...");
+        let guard = sentry::init((
+            dsn,
+            sentry::ClientOptions {
```
[the remark this answers is not in the document]
Comment (nerdypepper) [reply server/bleep/src/lib.rs:348]: sentry is only initialized in the `bleep` binary, and tauri only makes use of the library, so we don't end up doubly initializing sentry. 

the differentiation between dev and prod is made by changing the DSNs in the `.env` file as you said, the `environment` setting of the sentry client is automatically set to dev/prod based on `debug` or `release` builds.

[the remark this answers is not in the document]
Comment (ggordonhall) [reply server/bleep/src/lib.rs:348]: Gotcha. I don't think we have a dev DSN at all (we have separate DSNs for frontend and backend), and even in development we often run `bleep` in `release` mode. Is the solution here to setup another Sentry env?

[the remark this answers is not in the document]
Comment (nerdypepper) [reply server/bleep/src/lib.rs:348]: yes, i am leaning towards using a dev DSN to track events for dev deployments. we simply need to inject a new DSN into the dev container. local builds shouldn't set a DSN at all IMO, sentry can go uninitialized.

**Pull request merged**